src/Raven.Server/Documents/Replication/OutgoingDocumentReplication.cs

```diff
@@ -74,14 +78,20 @@ private void HandleDocumentChange(DocumentChangeNotification notification)
 
                 _waitForChanges.Reset();
 
-                try
-                {
+				var result = ReplicationBatchResult.NothingToDo;
+				try
+				{
                     _cancellationTokenSource.Token.ThrowIfCancellationRequested();
-                    await ExecuteReplicationOnce();
+	                result = await ExecuteReplicationOnce();
 
-                    if (_log.IsDebugEnabled)
+	                if (_log.IsDebugEnabled)
                         _log.Debug($"Finished replication for '{_replicationUniqueName}'.");
-                }
+
+					//if this is a transient result, don't retry immediately
+					//TODO: add here a back-off strategy?
```

Comment:
How do you know, and why do you care?
Errors always close the connection. The calling code will notice that and schedule another outgoing replication in a while.
All attempts to be smart with error handle should be avoided.